src/proxy/outbound.rs

```diff
@@ -512,6 +520,8 @@ struct Request {
     // The intended destination workload. This is always the original intended target, even in the case
     // of other proxies along the path.
     destination_workload: Option<Workload>,
+    destination_service_name: Option<String>,
```

Comment:
can we make this a Service object with name, namespace, and hostname? then we just have one

`destination_service: Option<Service>`

Reply:
Actually we don't need to make one, we already have Service in the `state` package. Unless it is hard to move around due to ownership - if so, then just make something like ServiceReference or whatever with those 3 fields